# Nama : Dzaky Athallah Subandi
# NIM  : 2211110040

In [2]:
# Import Library
import numpy as np
import pandas as pd
import re
import string
from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer, TfidfVectorizer

In [3]:
df = pd.read_csv('data.csv')
df = df.drop(columns='Unnamed: 0')
df.head()

userName  \
0               Gadri Bandu   
1  070_Putri Azzahra Nurdin   
2       Novrealita Setiyana   
3                auah gelap   
4               Adi Nugroho   

                                             content  score  
0  Saya kasih bintang tiga dulu soalnga gemenya b...      3  
1  Bagus bgtt. Dari segi story, grafik sampe puzz...      5  
2  Sejauh ini aku main aku suka banget sama model...      5  
3  Semuanya sudah sempurna tapi Sangat disayangka...      3  
4  Grafik bagus,story bagus,terutama archon quest...      5

In [4]:
text = df['content']
def hilang_angka_tandaBaca(text):
    clean = re.sub(r'[^a-zA-Z\s]', '', text)
    return clean
df['content']=text.apply(hilang_angka_tandaBaca)
print(df['content'])

0      Saya kasih bintang tiga dulu soalnga gemenya b...
1      Bagus bgtt Dari segi story grafik sampe puzzle...
2      Sejauh ini aku main aku suka banget sama model...
3      Semuanya sudah sempurna tapi Sangat disayangka...
4      Grafik bagusstory bagusterutama archon questpa...
                             ...                        
995    Masih bingung sih dengan cara game ini ngerend...
996    game nya seru dari segi story saran buat Hoyov...
997    Game ini sangat bagus dengan map yg sangat lua...
998    Sangat seru dan asik untuk dimainkan grafik hd...
999    Memuat game terlalu lama  Mau masuk game aja l...
Name: content, Length: 1000, dtype: object


In [5]:
tfidf = TfidfVectorizer(min_df=5)
tfidf_matrix = tfidf.fit_transform(df['content'])
print(tfidf_matrix)

  (0, 591)	0.14043768002372942
  (0, 466)	0.1947067952024529
  (0, 76)	0.2495192829470463
  (0, 161)	0.0780050289352639
  (0, 288)	0.08769497983329061
  (0, 421)	0.16555526061133186
  (0, 308)	0.18123585649316767
  (0, 633)	0.16918353099665973
  (0, 325)	0.13124173245415802
  (0, 217)	0.22868211749379186
  (0, 96)	0.1731060135288722
  (0, 571)	0.16469646780533265
  (0, 622)	0.22868211749379186
  (0, 504)	0.20040784577988735
  (0, 143)	0.07770174760073796
  (0, 371)	0.22479430550166687
  (0, 59)	0.11432981924493323
  (0, 462)	0.2377843999209768
  (0, 762)	0.23376019320758895
  (0, 525)	0.07651068045549124
  (0, 224)	0.13163213064254664
  (0, 73)	0.17643135083140213
  (0, 747)	0.2745619777948546
  (0, 789)	0.11740985926521082
  (0, 429)	0.1335512823419461
  :	:
  (997, 815)	0.13018339637531265
  (997, 399)	0.2171274165173858
  (997, 288)	0.1019964823964651
  (997, 143)	0.3614941218482225
  (997, 224)	0.07654950329776426
  (997, 47)	0.17011924410489945
  (998, 38)	0.47550162531576506
  (9

(0, 591) adalah representasi dari indeks dokumen dan indeks kata dalam matriks TF-IDF.

0 berarti dokumen ini merupakan dokumen pertama dalam dataset.

591 mengindikasikan bahwa ini adalah kata ke-591 (indeks dimulai dari 0) dalam matriks TF-IDF.

0.14043768002372942 adalah bobot TF-IDF dari kata dengan indeks 591 dalam dokumen pertama. Ini adalah nilai bobot yang menunjukkan pentingnya kata tersebut dalam dokumen pertama. Semakin tinggi bobotnya, semakin penting kata tersebut dalam konteks dokumen tersebut.

In [6]:
print(f'Jumlah fiturnya adalah {len(tfidf.get_feature_names_out())}.')
print(tfidf.get_feature_names_out())

Jumlah fiturnya adalah 818.
['abis' 'abyss' 'ada' 'adalah' 'adventure' 'agak' 'agar' 'air' 'aj' 'aja'
 'akan' 'akhir' 'akhirnya' 'aku' 'akun' 'alur' 'ama' 'aman' 'ampas' 'an'
 'anak' 'and' 'anda' 'android' 'aneh' 'animasi' 'anime' 'anniv'
 'anniversary' 'apa' 'apakah' 'apalagi' 'apapun' 'aplikasi' 'ar' 'archon'
 'arte' 'artefak' 'asik' 'atas' 'atau' 'awal' 'awalnya' 'ayo' 'backsound'
 'bagi' 'bagian' 'bagus' 'bagustapi' 'bahasa' 'bahkan' 'baik' 'bakal'
 'bakalan' 'balik' 'banget' 'bangett' 'banner' 'bansos' 'banyak'
 'banyakin' 'barang' 'bareng' 'baru' 'beberapa' 'beda' 'begini' 'begitu'
 'beli' 'belum' 'benar' 'bener' 'berasa' 'berat' 'berguna' 'berharap'
 'berhenti' 'beri' 'berikan' 'berkembang' 'bermain' 'bermasalah'
 'berpetualang' 'bertarung' 'besar' 'best' 'bet' 'betah' 'bgt' 'biar'
 'biasa' 'biasanya' 'bikin' 'bilang' 'bingung' 'bintang' 'bisa' 'blm'
 'bngt' 'boleh' 'boros' 'bos' 'bosan' 'bosen' 'bosenin' 'boss' 'btw'
 'buang' 'buat' 'bug' 'buka' 'bukan' 'bulan' 'buruk' 'but' 'b

In [7]:
final_data = pd.DataFrame(tfidf_matrix.toarray(), columns=tfidf.get_feature_names_out())
final_data

abis  abyss       ada  adalah  adventure      agak  agar  air   aj  \
0     0.0    0.0  0.000000     0.0        0.0  0.000000   0.0  0.0  0.0   
1     0.0    0.0  0.000000     0.0        0.0  0.000000   0.0  0.0  0.0   
2     0.0    0.0  0.057537     0.0        0.0  0.271517   0.0  0.0  0.0   
3     0.0    0.0  0.179229     0.0        0.0  0.000000   0.0  0.0  0.0   
4     0.0    0.0  0.000000     0.0        0.0  0.000000   0.0  0.0  0.0   
..    ...    ...       ...     ...        ...       ...   ...  ...  ...   
995   0.0    0.0  0.000000     0.0        0.0  0.000000   0.0  0.0  0.0   
996   0.0    0.0  0.000000     0.0        0.0  0.000000   0.0  0.0  0.0   
997   0.0    0.0  0.000000     0.0        0.0  0.000000   0.0  0.0  0.0   
998   0.0    0.0  0.000000     0.0        0.0  0.000000   0.0  0.0  0.0   
999   0.0    0.0  0.000000     0.0        0.0  0.000000   0.0  0.0  0.0   

          aja  ...     world  wow   ya  yaa  yah  yaitu      yang        yg  \
0    0.000000  ...  0.000000  0.0  0.0  0.0  0.0    0.0  0.000000  0.000000   
1    0.000000  ...  0.151562  0.0  0.0  0.0  0.0    0.0  0.000000  0.192304   
2    0.000000  ...  0.000000  0.0  0.0  0.0  0.0    0.0  0.000000  0.059198   
3    0.000000  ...  0.000000  0.0  0.0  0.0  0.0    0.0  0.000000  0.000000   
4    0.000000  ...  0.000000  0.0  0.0  0.0  0.0    0.0  0.080889  0.000000   
..        ...  ...       ...  ...  ...  ...  ...    ...       ...       ...   
995  0.000000  ...  0.000000  0.0  0.0  0.0  0.0    0.0  0.184185  0.000000   
996  0.157066  ...  0.453488  0.0  0.0  0.0  0.0    0.0  0.000000  0.143849   
997  0.000000  ...  0.000000  0.0  0.0  0.0  0.0    0.0  0.099376  0.130183   
998  0.000000  ...  0.000000  0.0  0.0  0.0  0.0    0.0  0.000000  0.000000   
999  0.221849  ...  0.000000  0.0  0.0  0.0  0.0    0.0  0.000000  0.000000   

     youtube  zhongli  
0        0.0      0.0  
1        0.0      0.0  
2        0.0      0.0  
3        0.0      0.0  
4        0.0      0.0  
..       ...      ...  
995      0.0      0.0  
996      0.0      0.0  
997      0.0      0.0  
998      0.0      0.0  
999      0.0      0.0  

[1000 rows x 818 columns]